jsonnet/telemeter/rules.libsonnet

```diff
@@ -160,6 +160,55 @@
+              // Only CoreOS nodes are counted.
+              // 1. x86_64 nodes that show hyperthreading in the telemetry have an accurate cores value in node_role_os_version_machine:cpu_capacity_cores:sum.
+              // 2. x86_64 nodes that are do not show hyperthreading and are bare metal have an accurate cores value in node_role_os_version_machine:cpu_capacity_cores:sum.
+              // 3. x86_64 nodes that are do not show hyperthreading and are not bare metal need the cores value adjusted to account for 2 threads per core (* 0.5).
+              // 4. Other CPU architectures are assumed to have accurate values in node_role_os_version_machine:cpu_capacity_cores:sum.
+              record: 'cluster:usage:workload:capacity_effective_cpu_cores',
+              expr: |||
+                # worker ht amd64
+                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_infra!="true",label_node_role_kubernetes_io_master!="true",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="true"}) or (0 * max by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum)))
+
+                +
+
+                # worker virt non-ht amd64
+                ((sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_infra!="true",label_node_role_kubernetes_io_master!="true",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="false"}) * on(_id) group(cluster_infrastructure_provider{type!="BareMetal"}) by (_id)) / 2.0 or (0 * max by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum)))
+
+                +
+
+                # worker phys non-ht amd64
+                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_infra!="true",label_node_role_kubernetes_io_master!="true",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="false"})* on(_id) cluster_infrastructure_provider{type="BareMetal"} or (0 * max by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum)))
+
+                +
+
+                # worker non-amd64
+                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_infra!="true",label_node_role_kubernetes_io_master!="true",label_kubernetes_io_arch!="amd64"}) or (0 * max by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum)))
+
+                +
+
+                # schedulable control plane ht amd64
+                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="true"})* on(_id) max by (_id) (cluster_master_schedulable) or (0 * max by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum)))
```

Comment:
unless I'm mistaken, it sums cores for all nodes while it should filter master nodes?

Reply:
Not certain... if we pick a cluster and do it with/without is it what we expect minus the control plane?

Reply:
Correct, I've been using a [python script](https://gist.github.com/kahowell/56368cd6aac4e3d2bb1176266c9d02a5) to generate the query for consistency, and it had a (now-fixed) typo in it on line 14 :disappointed:.